pkg/kubectl/cmd/cmd.go

```diff
@@ -145,6 +146,21 @@ func runHelp(cmd *cobra.Command, args []string) {
 	cmd.Help()
 }
 
+func calledByAlias(cmd *cobra.Command) bool {
+	if len(cmd.Aliases) == 0 {
+		return false
+	}
+	for _, arg := range os.Args {
+		if arg == cmd.Name() {
+			return false
+		}
+		if cmd.HasAlias(arg) {
```

Comment:
This fails when called under nested subcommands with potentially conflicting names, right?  `root subcommandConflictingWithAlias cmd.Name`.

I don't object to the intent, but given how commands can be used in multiple locations, I'd want to be pretty careful about how I implemented this.  Does cobra have some way of letting you know whether the `FindCommand` step succeeded on a name match or an alias match?